packages/codemod/src/upgrade-legacy.ts

```diff
@@ -1198,43 +1198,44 @@ const upgradeLegacy = async () => {
 
   steps.push({
     name: "check for usages of invokeWithMiddleware",
-    action: async (stepIndex) => {
-      let errors = 0
-
-      getAllFiles(appDir, [], [], [".ts", ".tsx", ".js", ".jsx"]).forEach((file) => {
-        const program = getCollectionFromSource(file)
-        const invokeWithMiddlewarePath = findCallExpression(program, "invokeWithMiddleware")
-        if (invokeWithMiddlewarePath?.length) {
-          log.error(`\n invokeWithMiddleware found at ${file}. \n`)
-          errors++
-        }
-      })
-
+    action: async () => {
       getAllFiles(path.resolve("pages"), [], [], [".ts", ".tsx", ".js", ".jsx"]).forEach((file) => {
         const program = getCollectionFromSource(file)
-        const invokeWithMiddlewarePath = findCallExpression(program, "invokeWithMiddleware")
-        if (invokeWithMiddlewarePath?.length) {
-          log.error(`\n invokeWithMiddleware found at ${file}. \n`)
-          errors++
+        try {
+          const invokeWithMiddlewarePath = findCallExpression(program, "invokeWithMiddleware")
+          if (invokeWithMiddlewarePath?.length) {
+            invokeWithMiddlewarePath.forEach((path) => {
+              path.value.arguments.push(j.identifier("ctx"))
+              const resolverExpression = j.callExpression(
+                //@ts-ignore
+                j.identifier(path.value.arguments[0]?.name),
+                path.value.arguments.slice(1),
+              )
+              const resolverStatement = j.expressionStatement(resolverExpression)
+              j(path).replaceWith(resolverStatement)
+            })              
+                      
+            const getServerSidePropsPath = findCallExpression(program, "gSSP")
+            if (getServerSidePropsPath) {
+              getServerSidePropsPath.forEach((path) => {
+                const {node} = path
+                const {arguments: asyncFunction} = node
+                //@ts-ignore
+                const params = asyncFunction[0]?.params
+                if (params && params.length) {
+                  params[0].properties.push(j.identifier("ctx"))
+                }
+              })
+            }
+          }
```

Comment:
Not sure we need this

Reply:
okay I will remove this.  
So should it be shown to the user that they need to get the ```ctx``` parameter from gSSP?

Reply:
> okay I will remove this. So should it be shown to the user that they need to get the `ctx` parameter from gSSP?

It should be expected that ctx is being passed already if they're using `invokeWithMiddleware`, as it's the 3rd argument.

Reply:
@dillonraphael I have removed this section